# Poisson distribution
The Poisson distribution and the binomial distribution have some similarities
but also several differences. The binomial distribution describes a distribution
of two possible outcomes designated as successes and failures from a given
number of trials. The Poisson distribution focuses only on the number of discrete occurrences over some interval or continuum. A Poisson experiment does not have a given number of trials (n) as a binomial experiment does.

**The Poisson distribution describes the occurrence of rare events. In fact, the Poisson formula has been referred to as the law of improbable events. For example, serious accidents at a chemical plant are rare, and the number per month might be described by the Poisson distribution.**

<h1>The Poisson distribution has the following characteristics:</h1>

■ It is a discrete distribution.

■ It describes rare events.

■ Each occurrence is independent of the other occurrences.

■ It describes discrete occurrences over a continuum or interval.

■ The occurrences in each interval can range from zero to infinity.

■ The expected number of occurrences must hold constant throughout the experiment.

<h2>Formula</h2>

$$P (x) =\frac{\lambda ^x e^\lambda}{x!}$$


where

$$x = 0, 1, 2, 3, . . .$$

$$\lambda = long -run ,average $$

$$e = 2.718282$$

In [2]:
import numpy as np
import pandas as pd
from scipy.stats import poisson as po

**Suppose bank customers arrive randomly on weekday afternoons at an average of 3.2
customers every 4 minutes. What is the probability of exactly 5 customers arriving in a
4-minute interval on a weekday afternoon?**

$$\lambda=3.2$$
$$x=5$$

In [7]:
prob=po.pmf(5,3.2)
print(round(prob,4))

0.114


probability of exactly 5 customers arriving in a 4-minute interval on a weekday afternoon is 1.14%

**Bank customers arrive randomly on weekday afternoons at an average of 3.2 customers
every 4 minutes. What is the probability of having more than 7 customers in
a 4-minute interval on a weekday afternoon?**

$$\lambda=3.2$$
$$x>7$$

In [9]:
prob=po.cdf(7,3.2)

In [10]:
prob_more_than_7=1-prob
print(round(prob_more_than_7,4))

0.0168


probability of of having more than 7 custumer in 4 minute is 1.68 %

**A bank has an average random arrival rate of 3.2 customers every 4 minutes. What
is the probability of getting exactly 10 customers during an 8-minute interval?**

$$\lambda=3.2 custumers/4 minutes$$
$$ x=10 custumers/10 minutes$$

$$ Hence$$

$$\lambda=6.4 custumers/8 minutes$$
$$ x=10 custumers/10 minutes$$


In [11]:
print(po.pmf(10,6.4))

0.052790043854115495


probability of getting exactly 10 customers during an 8-minute interval is 5.28 %

**The average number of annual trips per family to amusement parks in the United
States is Poisson distributed, with a mean of 0.6 trips per year.What is the probability of randomly selecting an American family and finding the following?**
* **The family did not make a trip to an amusement park last year.**
* **The family took exactly one trip to an amusement park last year.**
* **The family took two or more trips to amusement parks last year.**
* **The family took three or fewer trips to amusement parks over a three-year period.**
* **The family took exactly four trips to amusement parks during a six-year period.**


$$\lambda=0.6$$

In [16]:
#The family did not make a trip to an amusement park last year.
print(po.pmf(0,0.6))

0.5488116360940264


In [17]:
#The family took exactly one trip to an amusement park last year.
print(po.pmf(1,0.6))

0.3292869816564159


In [22]:
#The family took two or more trips to amusement parks last year.
prob=po.cdf(1,0.6)
print(round(1-prob,5))

0.1219


In [24]:
#The family took three or fewer trips to amusement parks over a three-year period.
# λ=0.6 per year hence λ=1.8 for three year

print(po.cdf(3,1.8))

0.8912916052907945


In [25]:
#The family took exactly four trips to amusement parks during a six-year period.
# λ=0.6 per year hence λ=3.6 for six year
print(po.pmf(4,3.6))

0.19122233917513215
